### PR TITLE
Api promises

### PR DESCRIPTION
**Pull request by AntonPuko**

### CODE REVIEW

File: src/dao/DAO.js [hunk 1]
```diff
@@ -220,36 +153,48 @@ class DAO<TModel, TModelMutator> {
...
-  async _resolve(action: ODataAction<TModel>): Promise<DAOResult<TModel>> {
+  async _resolve(
+    handler: oHandler<TModel>,
+    params: ?Object,
+    method: ?RequestMethod = 'get',
+  ): Promise<DAOResult<TModel>> {
     let request: Promise<ODataResult<TModel>>;
-    switch (action.method) {
+    switch (method) {
       case 'delete':
```
Comment (jlkalberer) [inline src/dao/DAO.js]: nit: can I get curly braces around these

File: interfaces/odata.js [hunk 2]
```diff
@@ -19,8 +19,8 @@ declare module 'odata' {
   };
 
   declare type OHandler<TModel> = {
```
Comment (jlkalberer) [inline interfaces/odata.js]: Do we still even need this?

File: interfaces/odata.js [hunk 3]
```diff
@@ -19,8 +19,8 @@ declare module 'odata' {
...
-     inlineCount: number,
```
Comment (jlkalberer) [inline interfaces/odata.js]: I would prefer `inlineCount`

File: interfaces/odata.js (hunk 2 above)
Comment (AntonPuko) [inline interfaces/odata.js]: I don't like it, but at this point, we still use it here: https://github.com/Brewskey/brewskey.admin/blob/dev/app/auth/sagas.js#L46-L56

File: interfaces/odata.js (hunk 3 above)
Comment (AntonPuko) [inline interfaces/odata.js]: I too, but this is how it is inside o.js.. it has `inlineCount()` to add counting to request, and `inlinecount` as result..we probably need to change it inside our o.js fork to `count`  for result

Comment (jlkalberer) [inline interfaces/odata.js]: Yeah, `count` is fine.

Comment (AntonPuko) [inline interfaces/odata.js]: eh.. o.js has `count` and `inlineCount` methods and `inlinecount` var. so I can't rename `inlinecount` to `count`

**Pull request merged**